File: Algorithms/src/761. Special Binary String/Code/solution.js
/**
 * @param {string} s
 * @return {string}
 */
var makeLargestSpecial = function (s) {
  let parts = [];
  let count = 0;
  let start = 0;

  for (let i = 0; i < s.length; i++) {
    if (s[i] === "1") count++;
    else count--;

    if (count === 0) {
      let inner = makeLargestSpecial(s.substring(start + 1, i));
      parts.push("1" + inner + "0");
      start = i + 1;
    }
  }

  // Sort descending
  parts.sort((a, b) => b.localeCompare(a));

  return parts.join("");
};

File: Algorithms/src/761. Special Binary String/Code/solution.py
class Solution:
    def makeLargestSpecial(self, s: str) -> str:
        parts = []
        count = 0
        start = 0
        
        for i, ch in enumerate(s):
            if ch == '1':
                count += 1
            else:
                count -= 1
            
            if count == 0:
                # Recursively solve inner substring
                inner = self.makeLargestSpecial(s[start+1:i])
                parts.append("1" + inner + "0")
                start = i + 1
        
        # Sort in descending order
        parts.sort(reverse=True)
        
        return "".join(parts)